crates/aptos-runtimes/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use std::sync::atomic::{AtomicUsize, Ordering};
+use tokio::runtime::{Builder, Runtime};
+
+/// The max thread name length before the name will be truncated
+/// when it's displayed. Note: the max display length is 15, but
+/// we need to leave space for the thread IDs.
+const MAX_THREAD_NAME_LENGTH: usize = 12;
+
+/// Returns a tokio runtime with named threads.
+/// This is useful for tracking threads when debugging.
+pub fn spawn_named_runtime(thread_name: String, num_worker_threads: Option<usize>) -> Runtime {
+    // Verify the given name has an appropriate length
+    if thread_name.len() > MAX_THREAD_NAME_LENGTH {
+        panic!(
```

Comment:
Yeah, I thought about this, but there's a couple issues:
1. Runtimes should only ever be spawned on startup. I'm not aware of use-cases that spawn runtimes after the node is running?
2. Even so, if this fails to spawn the runtime and returns an error, what happens next? My guess would be that all callers will simply panic anyway because now some part of the code that should be running isn't, and well, we can't keep going?
3. If this check fails, we could log an error, but to be honest, logs don't get bubbled up and we probably wouldn't notice until we actually need to debug the process, which is too late?

So I figured this is probably the best we could do for now?

Reply:
I think so. I but thought since this is a "common" crate, we would want to leave the decision to panic or not to the caller.

Reply:
Gotcha. Yeah, I think we can leave it as is for now, and then modify it in the future if we find a use case that doesn't panic. But, I'd be surprised 😄